projects/valtimo/components/src/lib/services/cds-theme.service.ts

```diff
@@ -46,31 +53,53 @@ export class CdsThemeService implements OnDestroy {
   }
 
   private openDarkModeSubscription(): void {
-    combineLatest([this.getBrowserPrefersDarkModeObservable(), this.preferredTheme$]).subscribe(
-      ([browserPrefersDarkMode, preferredTheme]) => {
-        switch (preferredTheme) {
-          case SelectableCarbonTheme.SYSTEM:
-            if (browserPrefersDarkMode) {
-              this.setThemeOnDocument(SelectableCarbonTheme.G90);
-            } else {
-              this.setThemeOnDocument(SelectableCarbonTheme.G10);
-            }
-            break;
-          case SelectableCarbonTheme.G10:
-            this.setThemeOnDocument(SelectableCarbonTheme.G10);
-            break;
-          case SelectableCarbonTheme.G90:
-            this.setThemeOnDocument(SelectableCarbonTheme.G90);
-            break;
-        }
-      }
+    this._subscriptions.add(
+      this.preferredTheme$
+        .pipe(
+          switchMap((preferredTheme: SelectableCarbonTheme) =>
+            preferredTheme === SelectableCarbonTheme.SYSTEM
+              ? this.getBrowserPrefersDarkModeObservable()
+              : of(preferredTheme)
+          )
+        )
+        .subscribe((theme: SelectableCarbonTheme | boolean) => {
+          let documentTheme: SelectableCarbonTheme;
+          let editorTheme: SelectableMonacoTheme;
+
+          if (isBoolean(theme)) {
+            documentTheme = theme ? SelectableCarbonTheme.G90 : SelectableCarbonTheme.G10;
+            editorTheme = theme ? SelectableMonacoTheme.VSDARK : SelectableMonacoTheme.VS;
+          } else {
+            documentTheme = theme;
+            editorTheme =
+              theme === SelectableCarbonTheme.G10
+                ? SelectableMonacoTheme.VS
+                : SelectableMonacoTheme.VSDARK;
+          }
+          this.setThemeOnDocument(documentTheme);
+          this._editorTheme$.next(editorTheme);
+        })
+    );
+  }
+
+  private openMonacoThemeSubscription(): void {
```

Comment:
This logic should be moved to the EditorService, where also the editor instance should be saved